kitsune/questions/classifiers.py

```diff
@@ -0,0 +1,43 @@
+from langchain.schema.runnable import RunnableLambda, RunnablePassthrough
+
+from kitsune.llm.questions.prompt import spam_parser, spam_prompt, topic_parser, topic_prompt
+from kitsune.llm.utils import get_llm
+from kitsune.products.utils import get_taxonomy
+from kitsune.questions.models import Question
+
+DEFAULT_LLM_MODEL = "gemini-2.5-flash-preview-04-17"
+
+
+def analyze_question(question: Question) -> tuple[bool, dict]:
+    """
+    Analyze a question from spam and classifies the topic.
+    """
+    llm = get_llm(model_name=DEFAULT_LLM_MODEL)
+
+    product = question.product
+    payload = {
+        "question": question,
+        "product": product,
+        "topics": get_taxonomy(product),
+    }
+
+    spam_detection_chain = spam_prompt | llm | spam_parser
+    topic_classification_chain = topic_prompt | llm | topic_parser
+
+    pipeline = RunnablePassthrough.assign(spam_result=spam_detection_chain) | RunnableLambda(
+        lambda payload: {
+            **payload["spam_result"],
+            **(
+                {
+                    "topic_result": (
+                        None
+                        if payload["spam_result"]["is_spam"]
+                        else topic_classification_chain.invoke(payload)
+                    )
+                }
+            ),
+        }
+    )
```

Comment:
For me, this is difficult to read due to the `LangChain` abstractions. Since `LangChain` automatically wraps Python callables with `RunnableLambda`, what do you think of something like the following instead? I've tested it and it returns the same result. For me, it feels more "Pythonic" and readable. 

```python
    def conditional_topic_classification_chain(result):
        if result["is_spam"] or (result["confidence"] < 60):
            result.update(topic_result=None)
        else:
            result.update(topic_result=topic_classification_chain.invoke(payload))
        return result

    pipeline = spam_detection_chain | conditional_topic_classification_chain
```